src/screens/SignupScreen/index.tsx

```diff
@@ -1,85 +1,106 @@
-import { useNavigation } from '@react-navigation/native';
-import { NativeStackNavigationProp } from '@react-navigation/native-stack';
-import React, { FC, useState } from 'react';
-import { View, Text, KeyboardAvoidingView, TouchableOpacity } from 'react-native';
-import Ionicons from '@expo/vector-icons/Ionicons';
+import { useNavigation } from "@react-navigation/native";
+import { NativeStackNavigationProp } from "@react-navigation/native-stack";
+import React, { FC, useRef, useState } from "react";
+import {
+  View,
+  Text,
+  KeyboardAvoidingView,
+  TouchableOpacity,
+  TextInput,
+} from "react-native";
+import Ionicons from "@expo/vector-icons/Ionicons";
 
-import { AuthRootView, Input, Button } from '../../components';
-import { AuthStackParamList } from '../../navigation/AuthNavigator';
-import { isIOS } from '../../utils';
+import { AuthRootView, Input, Button } from "../../components";
+import { AuthStackParamList } from "../../navigation/AuthNavigator";
+import { isIOS } from "../../utils";
 
-import styles from './styles';
-import { StatusBar } from 'expo-status-bar';
-import { getAuth, createUserWithEmailAndPassword } from 'firebase/auth';
+import styles from "./styles";
+import { StatusBar } from "expo-status-bar";
+import { getAuth, createUserWithEmailAndPassword } from "firebase/auth";
 
-type TNavigationProp = NativeStackNavigationProp<AuthStackParamList, 'SignupScreen'>;
+type TNavigationProp = NativeStackNavigationProp<
+  AuthStackParamList,
+  "SignupScreen"
+>;
 
 const SignupScreen: FC = () => {
-  const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const passwordInput = useRef<TextInput>(null!);
 
   const navigation = useNavigation<TNavigationProp>();
 
   const onEmailPasswordSignup = async () => {
     try {
       const auth = getAuth();
-      await createUserWithEmailAndPassword(auth, email, password)
-    } catch(e) {
+      await createUserWithEmailAndPassword(auth, email, password);
+    } catch (e) {
       console.warn(e);
     }
   };
 
   return (
     <AuthRootView backgroundTitle={"Social"}>
       <StatusBar style="light" />
-      <View style={styles.topHalfContainer}/>
-        <KeyboardAvoidingView
-          behavior={isIOS() ? 'padding' : 'height'}
-          style={styles.bottomHalfContainer}
+      <View style={styles.topHalfContainer} />
+      <KeyboardAvoidingView
+        behavior={isIOS() ? "padding" : "height"}
+        style={styles.bottomHalfContainer}
+      >
+        <Text style={styles.subtitleText}>{"Sign Up"}</Text>
+        <Text style={styles.descriptionText}>
+          {"Choose how you'd like to sign up."}
+        </Text>
+        <Input
+          onChangeText={setEmail}
+          value={email}
+          textInputProps={{
+            returnKeyType: "next",
+            blurOnSubmit: false,
```

Comment:
I was stuck on this for a bit, as I needed an `onBlur` function to trigger while still maintaining the keyboard. `onSubmitEditing` with `blurOnSubmit` set to false was my answer. `onSubmitEditing` will trigger while `blurOnSubmit` stops the Keyboard from auto-dismissing.

Unrelated, but funnily enough, `blurOnSubmit={false}` still allows the internal `onBlur` function to trigger!